tests/test_integration.py

```diff
@@ -104,6 +104,8 @@ def setUp(self):
 
         # Cleanup all leftovers from previous tests or failures.
         clean_all(self.route_network, self.datacenter_type, VM_HOSTNAME)
+        if self.datacenter_type == 'aws.dct':
```

Comment:
any reason this should not go into `clean_all` function?

Reply:
clean_all gets called more than once afais, so it will remove testing VMs, which are still in use for testing.

Reply:
ah now understand. i maybe have a better idea in general to solve this. the actual problem is that SA objects might be missing that hold the instance_id right? so instead of making a whole new code branch we could instead solve only this one edge case. in `clean_aws()` there is this `try` block which catches the `DatasetError`. i think what we might really want to do is instead of returning, try to find the right instance id through aws api. what do you think? then we do not have to work around this and can keep all existing logic int `clean_all()` and `clean_aws()`

Reply:
good idea, let's see, how this goes.